### PR TITLE
#714: Use the build cache to speed up the LS

### PR DESCRIPTION
**Pull request by kittaakos**
### Motivation

Avoid reindexing libraries on every single text document change event.

### Change description

After a verify (both success and failure), the IDE2 calls the `arduino.languageserver.notifyBuildDidComplete` command (exposed by the VS Code extension) and pushes the most recent [`build_path`](https://github.com/arduino/arduino-cli/blob/63b53c0fa6830ce77cfb5508f06b2b4e601c90b5/rpc/cc/arduino/cli/commands/v1/compile.proto#L44) to the LS via the custom `ino/buildDidComplete` JSON-RPC.

 - Use `0.25.0-rc1` CLI,
 - Use `0.0.2-beta.4` VS Code extension, and
 - Use `0.7.1` Arduino LS

Dev:
 - Updated the build scripts; can build the LS from sources besides downloading it from S3 or GH.

### Other information

#714 

### Reviewer checklist

* [ ] PR addresses a single concern.
* [ ] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [ ] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: arduino-ide-extension/src/browser/contributions/core-error-handler.ts [hunk 1]
```diff
@@ -4,29 +4,29 @@ import { CoreError } from '../../common/protocol/core-service';
...
-    CoreError.Compiler[]
+    CoreError.ErrorLocation[]
   >();
 
   tryHandle(error: unknown): void {
     if (CoreError.is(error)) {
-      this.compilerErrors.length = 0;
-      this.compilerErrors.push(...error.data.filter(CoreError.Compiler.is));
+      this.errors.length = 0;
+      this.errors.push(...error.data);
```
Comment (fstasi) [inline arduino-ide-extension/src/browser/contributions/core-error-handler.ts:15]: Isn't this equivalent to
```suggestion
      this.errors = error.data;
```
performance wise is also faster

File: electron/build/template-package.json [hunk 2]
```diff
@@ -141,7 +141,7 @@
   "theiaPluginsDir": "plugins",
   "theiaPlugins": {
     "vscode-builtin-cpp": "https://open-vsx.org/api/vscode/cpp/1.52.1/file/vscode.cpp-1.52.1.vsix",
-    "vscode-arduino-tools": "https://downloads.arduino.cc/vscode-arduino-tools/vscode-arduino-tools-0.0.2-beta.2.vsix",
+    "vscode-arduino-tools": "https://github.com/kittaakos/vscode-arduino-tools/raw/realTimeDiagnostics/build-artifacts/vscode-arduino-tools-0.0.2-beta.3.vsix",
```
Comment (fstasi) [inline electron/build/template-package.json:144]: can we rely on https://github.com/arduino/vscode-arduino-tools ?

File: package.json [hunk 3]
```diff
@@ -75,7 +75,7 @@
   "theiaPluginsDir": "plugins",
   "theiaPlugins": {
     "vscode-builtin-cpp": "https://open-vsx.org/api/vscode/cpp/1.52.1/file/vscode.cpp-1.52.1.vsix",
-    "vscode-arduino-tools": "https://downloads.arduino.cc/vscode-arduino-tools/vscode-arduino-tools-0.0.2-beta.2.vsix",
```
Comment (fstasi) [inline package.json:78]: same as before, I'd rather go with https://github.com/arduino/vscode-arduino-tools if possible

**Review by fstasi — CHANGES_REQUESTED** [review]
lgtm, only minor concerns on using kittaakos fork for the `vscode-arduino-tools`

File: arduino-ide-extension/src/browser/contributions/core-error-handler.ts (hunk 1 above)
Comment (kittaakos) [reply arduino-ide-extension/src/browser/contributions/core-error-handler.ts:15]: The fields are `readonly`. Your proposal is a compiler error. Did I overlook something?

File: electron/build/template-package.json (hunk 2 above)
Comment (kittaakos) [reply electron/build/template-package.json:144]: Sure sure. I did not want to create an RC before testing it out.

**Comment by kittaakos** [comment]
The PR pinned the version of the VS Code extension and the LS. @fstasi, please review and merge the PR if you are happy with it. Thank you!

**Review by fstasi — APPROVED** [review]

**Pull request merged**